src/util/url.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @flow
+ * @file Utility functions for urls
+ * @author Box
+ */
+import Uri from 'jsuri';
+
+/**
+ * Update URL query parameters
+ *
+ * @param {string} url - the url that contains the potential query parameter string
+ * @param {Object} queryParams
+ * @return {string}
+ */
+function updateQueryParameters(url: string, queryParams: Object): string {
```

Comment:
Can sessionEndpoints' values all be of type Uri instead of String, so that we can just call addQueryParam/replaceQueryParam directly?

Reply:
we only update the query params for commit and listParts, not sure if it's worth it. I'll consider it when we find ourselves messing with the urls more